ZipBuilder/Sources/FirebaseReleaser/Tags.swift

```diff
@@ -36,12 +36,43 @@ enum Tags {
         fatalError("Non-Firebase pod \(pod.name) is missing a version")
       }
       let tag = pod.name.replacingOccurrences(of: "Google", with: "") + "-" + version
-      createTag(gitRoot: gitRoot, tag: tag)
+      createTag(gitRoot: gitRoot, tag: tag, afterDelete: afterDelete)
     }
   }
 
-  private static func createTag(gitRoot: URL, tag: String) {
+  static func updateTags(gitRoot: URL) {
+    create(gitRoot: gitRoot, afterDelete: true)
+  }
+
+  private static func createTag(gitRoot: URL, tag: String, afterDelete: Bool) {
+    if afterDelete {
+      verifyDeleteIsOK()
+      Shell.executeCommand("git tag --delete \(tag)", workingDir: gitRoot)
+      Shell.executeCommand("git push --delete origin \(tag)", workingDir: gitRoot)
+    }
     Shell.executeCommand("git tag \(tag)", workingDir: gitRoot)
     Shell.executeCommand("git push origin \(tag)", workingDir: gitRoot)
   }
+
+  private static func verifyDeleteIsOK() {
+    var homeDirURL: URL
+    if #available(OSX 10.12, *) {
+      homeDirURL = FileManager.default.homeDirectoryForCurrentUser
+    } else {
+      fatalError("Run on at least macOS 10.12")
+    }
+    let manifest = FirebaseManifest.shared
+    let firebasePublicURL = homeDirURL.appendingPathComponents(
+      [".cocoapods", "repos", "cocoapods", "Specs", "0", "3", "5", "Firebase"]
```

Comment:
I curious where the number components come from?

Reply:
It's the CocoaPods hash of the word Firebase. It's how CocoaPods does sharding.